//
// Copyright 2018 Asylo authors
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

syntax = "proto2";

package hello_world;

import "asylo/enclave.proto";
import "src/proto/capsule.proto";

// Extension to EnclaveInput for use with HelloWorld enclave.
message HelloInput {
  optional string to_greet = 1;
}

message HelloOutput {
  optional string greeting_message = 1;
}


message EnclaveResponder {
  optional int64 responder = 1;
}
message HotMsg {
  optional int64 buffer = 1;
}

extend asylo.EnclaveLoadConfig {
  optional EnclaveResponder enclave_r = 8086; 
}

extend asylo.EnclaveInput {
  optional HelloInput enclave_input_hello = 8086;
  optional CapsulePDU input_dc = 8087;
  optional HotMsg buffer = 8088;
  optional EnclaveResponder enclave_responder = 8089; 

}

extend asylo.EnclaveOutput {
  optional HelloOutput enclave_output_hello = 8088;
  optional CapsulePDU output_dc = 8089;
}
